test/metabase/query_processor_test/expressions_test.clj

```diff
@@ -143,14 +143,20 @@
 (deftest expressions-should-include-type-test
   (mt/test-drivers (mt/normal-drivers-with-feature :expressions)
     (testing "Custom aggregation expressions should include their type"
-      (is (= (conj #{{:name "x" :base_type (:base_type (qp.test/aggregate-col :sum :venues :price))}}
-                   {:name      (mt/format-name "category_id")
-                    :base_type (:base_type (qp.test/breakout-col :venues :category_id))})
-             (set (map #(select-keys % [:name :base_type])
-                       (mt/cols
-                         (mt/run-mbql-query venues
-                           {:aggregation [[:aggregation-options [:sum [:* $price -1]] {:name "x"}]]
-                            :breakout    [$category_id]})))))))))
+      (let [cols (mt/cols
+                  (mt/run-mbql-query venues
+                    {:aggregation [[:aggregation-options [:sum [:* $price -1]] {:name "x"}]]
+                     :breakout    [$category_id]}))]
+        (testing (format "cols = %s" (u/pprint-to-str cols))
+          (is (= #{"x" (mt/format-name "category_id")}
+                 (set (map :name cols))))
+          (let [name->base-type (into {} (map (juxt :name :base_type) cols))]
+            (testing "x"
+              (is (isa? (name->base-type "x")
+                        :type/Number)))
+            (testing "category_id"
+              (is (isa? (name->base-type (mt/format-name "category_id"))
+                        :type/Number)))))))))
```

Comment:
Different DBs return different base types for the breakout column; as long as it's a `:type/Number` of some sort that's all we really care about